app/gui/src/dashboard/components/JSONSchemaInput.tsx

```diff
@@ -186,7 +200,7 @@ export default function JSONSchemaInput(props: JSONSchemaInputProps) {
               >
                 {propertyDefinitions.map((definition) => {
                   const { key, schema: childSchema } = definition
-                  const isOptional = !requiredProperties.includes(key)
+                  const isOptional = !requiredProperties.includes(key) || isAbsent
```

Comment:
I think this may be too simple for the fix we need.

Fields that are themselves required, should still be marked red and not accepted if they are absent.

Only required **sub-fields** whose _parent field is optional_, do not have to be required.

Reply:
i guess this is a bit of a misnomer - `isAbsent` means that the parent field is optional

Reply:
```
Foo (object field,  optional, currently absent, isAbsent = false)
  Bar (isAbsent = true, not optional)
  Baz (isAbsent = true, not optional)
```

Reply:
> i guess this is a bit of a misnomer - `isAbsent` means that the parent field is optional

Oh... Yeah so that name is a bit misleading indeed